doc/tools/coqrst/coqdomain.py

```diff
@@ -1143,6 +1155,33 @@ def GrammarProductionRole(typ, rawtext, text, lineno, inliner, options={}, conte
 
 GrammarProductionRole.role_name = "production"
 
+
+def GlossaryDefRole(typ, rawtext, text, lineno, inliner, options={}, content=[]):
+    """Marks the definition of a glossary term inline in the text.  Matching :term:`XXX`
+    constructs will link to it.  The term will also appear in the Glossary Index.
+
+    Example::
+
+       A :gdef:`prime` number is divisible only by itself and 1.
+    """
+    #pylint: disable=dangerous-default-value, unused-argument
+    env = inliner.document.settings.env
+    std = env.domaindata['std']['objects']
+    key = ('term', text)
+
+    if key in std:
+        MSG = 'Duplicate object: {}; other is at {}'
+        msg = MSG.format(text, env.doc2path(std[key][0]))
+        inliner.document.reporter.warning(msg, line=lineno)
+
+    targetid = 'term-{}'.format(text.replace(' ', '-'))
+    std[key] = (env.docname, targetid)
+    target = nodes.target('', '', ids=[targetid], names=[text])
+    node = nodes.inline(rawtext, '', target, nodes.Text(text), classes=['term-defn'])
```

Comment:
Are you missing a `note_explicit_target` here? (As well as a `set_role_source_info` maybe?)

Reply:
No idea what those do or why they might be desirable, but the code seems to work the same with them.